knora/dsp_tools.py

```diff
@@ -133,7 +133,10 @@ def program(user_args: list[str]) -> None:
                              dump=args.dump)
         else:
             if args.validate:
-                validate_ontology(args.datamodelfile)
+                if validate_ontology(args.datamodelfile):
+                    exit(0)
+                else:
+                    exit(1)
```

Comment:
makes sense like this.

Additionally, I think, similar handling should happen on:
- line 120 (exit code `1` not  `0`)
- line 125?

Is it correct that there is no `xml_upload` with validation only?

Reply:
Makes sense, I was only focussing on the ontology and xml, but you are right of course. I added it on both places.

For `xml_upload` it's implemented a bit differently with a separate parameter to the function called `validate_only`.